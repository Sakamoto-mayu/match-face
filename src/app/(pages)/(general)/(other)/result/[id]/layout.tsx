export const metadata = {
  title: "回答確認",
  description: "一般ユーザー回答確認ページ",
};

export default function ResultLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
