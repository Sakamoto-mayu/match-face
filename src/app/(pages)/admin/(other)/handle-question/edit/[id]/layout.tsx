export const metadata = {
    title: "質問・解答例編集",
    description: "管理者質問・解答例編集ページ",
  };
  
  export default function editLayout({
    children,
  }: {
    children: React.ReactNode;
  }) {
    return <>{children}</>;
  }
  