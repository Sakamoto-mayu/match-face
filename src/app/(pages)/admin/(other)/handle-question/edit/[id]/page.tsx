"use client";
import answerEditData from "@/const/answerEdit";
import Input from "@/components/ui/Input";
import { useState } from "react";
import OrangeButton from "@/components/ui/button/OrangeButton";
import WhiteButton from "@/components/ui/button/WhiteButton";
import Link from "next/link";
import type { ChangeEvent } from "react";
import type { Question } from "@/const/projectTable";

const EditQuestionPage = () => {
  const data = answerEditData[0];
  const [editData, setEditData] = useState<Question[]>(
    answerEditData[0].questions
  );
  const [newID, setNewId] = useState<number>(editData.length + 1);
  const [errMsg, setErrMsg] = useState("");

  const addWriteQuestion = () => {
    setEditData([
      ...editData,
      {
        question_id: newID,
        question: "",
        answer_example: "",
        answer: "",
        choices: [],
        select: false,
      },
    ]);
    setNewId(newID + 1);
  };

  const addSelectQuestion = () => {
    setEditData([
      ...editData,
      {
        question_id: newID,
        question: "",
        answer_example: "",
        answer: "",
        choices: ["", "", "", ""],
        select: true,
      },
    ]);
    setNewId(newID + 1);
  };

  const handleQuestionDelete = (index: number) => {
    const updatedData = [...editData];
    updatedData.splice(index, 1);
    setEditData(updatedData);
  };

  const sendData = () => {
    editData.map((data) => {
      if (data.select === false) {
        if (data.answer_example === "" || data.question === "") {
          setErrMsg("未入力の項目があります");
          return;
        }
      } else {
        if (
          data.answer === "" ||
          data.choices.includes("") ||
          data.question === ""
        ) {
          setErrMsg("未入力の項目があります");
          return;
        }
      }
    });
    if (errMsg !== "") {
      return;
    }
    console.log(editData);
  };

  const handleQuestionChange = (
    e: ChangeEvent<HTMLTextAreaElement>,
    index: number
  ) => {
    const updatedData = [...editData];
    updatedData[index].question = e.target.value;
    setEditData(updatedData);
  };

  const handleAnswerExampleChange = (
    e: ChangeEvent<HTMLTextAreaElement>,
    index: number
  ) => {
    const updatedData = [...editData];
    updatedData[index].answer_example = e.target.value;
    setEditData(updatedData);
  };

  const handleChoiceChange = (
    e: ChangeEvent<HTMLInputElement>,
    questionIndex: number,
    choiceIndex: number
  ) => {
    const updatedData = [...editData];
    if (
      updatedData[questionIndex]?.choices !== undefined &&
      updatedData[questionIndex].choices !== undefined
    ) {
      updatedData[questionIndex].choices[choiceIndex] = e.target.value;
      setEditData(updatedData);
    }
  };

  const handleAnswerChange = (
    e: ChangeEvent<HTMLSelectElement>,
    index: number
  ) => {
    const updatedData = [...editData];
    updatedData[index].answer = e.target.value;
    setEditData(updatedData);
  };
  return (
    <div className="flex flex-col items-center my-20 w-4/5 mx-auto h-auto">
      <div className="w-2/3">
        <div className="flex items-end justify-between border-b-2 border-black pb-3">
          <h1 className="text-3xl">質問・回答例の編集</h1>
          <p>登録日:{data.edit_date}</p>
        </div>
        <div className="my-8">
          <h2 className="text-2xl">案件名：{data.project_name}</h2>
          <p className="border p-3 mt-2">{data.project_detail}</p>
        </div>
      </div>
      {/* <QuestionList questions={questions} /> */}
      {editData.map((data, index) => {
        if (data.select === false) {
          return (
            <div
              key={data.question_id}
              className="flex flex-col w-2/3 my-5 border-b border-black pb-16 border-dashed"
            >
              <h2
                className="text-2xl"
                data-testid={`write_${data.question_id}`}
              >
                Q{index + 1}.
              </h2>
              <textarea
                name="question"
                id="question"
                value={data.question}
                className="border-deep-gray border  p-1 mt-2"
                rows={5}
                onChange={(e) => handleQuestionChange(e, index)}
                data-testid={`question_${data.question_id}`}
              ></textarea>
              <label htmlFor="answer_ex" className="my-3 text-2xl">
                回答例
              </label>
              <textarea
                name="answer_ex"
                id="answer_ex"
                value={data.answer_example}
                className="border-deep-gray border p-1"
                rows={5}
                onChange={(e) => handleAnswerExampleChange(e, index)}
                data-testid={`answer_ex_${data.question_id}`}
              ></textarea>
              <WhiteButton
                label="削除"
                className="w-20 text-xs py-1 mx-auto mt-5"
                onClick={() => handleQuestionDelete(index)}
              />
            </div>
          );
        } else if (data.select === true) {
          return (
            <div
              key={data.question_id}
              className="flex flex-col w-2/3 my-5 border-b border-black pb-16 border-dashed"
            >
              <h2
                className="text-2xl"
                data-testid={`select_${data.question_id}`}
              >
                Q{index + 1}.
              </h2>
              <textarea
                name="question"
                id="question"
                value={data.question}
                className="border-deep-gray border p-1 mt-2"
                rows={5}
                onChange={(e) => handleQuestionChange(e, index)}
                data-testid={`question_${data.question_id}`}
              ></textarea>
              {data.choices?.map((select, choiceIndex) => {
                return (
                  <div key={choiceIndex} className="flex mt-5">
                    <label htmlFor={`select${choiceIndex}`}>
                      選択肢{choiceIndex + 1}:
                    </label>
                    <Input
                      id={`select${choiceIndex}`}
                      value={select}
                      key={choiceIndex}
                      className="border-deep-gray pl-1"
                      onChange={(e) =>
                        handleChoiceChange(e, index, choiceIndex)
                      }
                      data-testid={`choice_${choiceIndex + 1}`}
                    />
                  </div>
                );
              })}
              <div className="flex mt-5">
                <label htmlFor="answer">回答:</label>
                <select
                  name="answer"
                  id="answer"
                  className="border-deep-gray"
                  defaultValue={data.answer}
                  onChange={(e) => handleAnswerChange(e, index)}
                >
                  <option value="">選択してください</option>
                  {data.choices?.map((select, index) => {
                    return (
                      <option
                        value={`${select}`}
                        key={index}
                        className="border-deep-gray"
                      >
                        {select}
                      </option>
                    );
                  })}
                </select>
              </div>
              <WhiteButton
                label="削除"
                className="w-20 text-xs py-1 mx-auto mt-5"
                onClick={() => handleQuestionDelete(index)}
              />
            </div>
          );
        }
      })}
      <div className="flex flex-col">
        <div className="flex justify-around m-3">
          <WhiteButton
            label="記述質問追加"
            className="m-10"
            onClick={addWriteQuestion}
            data-testid="addWriteButton"
          />
          <WhiteButton
            label="選択質問追加"
            className="m-10"
            onClick={addSelectQuestion}
            data-testid="addSelectButton"
          />
        </div>
        <p className="text-center text-red" data-testid="errMsg">
          {errMsg}
        </p>
        <div className="flex m-3">
          <Link href={`/admin/handle-question`} data-testid="backList">
            <OrangeButton
              label="< 一覧へ戻る"
              className="m-10 rounded-none py-5 flex items-center justify-center"
            />
          </Link>
          <OrangeButton
            label="保存する"
            className="m-10 rounded-none py-5 flex items-center justify-center"
            onClick={sendData}
            data-testid="sendButton"
          />
        </div>
      </div>
    </div>
  );
};

export default EditQuestionPage;
