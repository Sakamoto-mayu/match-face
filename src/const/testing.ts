export const project = [
  {
    id: "1",
    name: "案件のタイトルが入ります",
    date: new Date("2023/07/01"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
  {
    id: "2",
    name: "案件のタイトルが入ります",
    date: new Date("2023-07-02"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
  {
    id: "3",
    name: "案件のタイトルが入ります",
    date: new Date("2023-07-03"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
];

export const question = [
  {
    id: "1",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "2",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "3",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "4",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "5",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "6",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "7",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "8",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "9",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
];
