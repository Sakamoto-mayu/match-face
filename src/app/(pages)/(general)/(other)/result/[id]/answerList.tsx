import { question } from "@/const/testing";
import { Answer } from "@/const/result";
import ModelAnswerContent from "@/components/pages/general/result/ModelAnswer";

type Props = {
  project_id: number;
};

export const AnswerList = ({ project_id }: Props) => {
  const currentQuestion = question.filter(
    (question) => question.project_id === project_id
  );

  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === project_id
  );

  return (
    <>
      {currentQuestion.map((question, index) => (
        <div key={index} className="py-5">
          <h3 className="text-xl">
            Q{index + 1}:{question.name}
          </h3>
          <p id={`content-${index + 1}`} className="mt-2 mb-4">
            {question.content}
          </p>
          {question.type === "writing" ? (
            <div
              id={`answer-${index + 1}`}
              className="border border-gray p-3 mt-2 rounded-md shadow-md"
            >
              <h3 className="mb-2 text-xl">あなたの回答</h3>
              {currentAnswer[index].content}
            </div>
          ) : (
            <>
              <ul>
                {question.choices?.map((choice, index) => (
                  <li key={index}>・{choice}</li>
                ))}
              </ul>
              <div
                id={`answer-${index + 1}`}
                className="border border-gray p-3 mt-5 rounded-md shadow-md"
              >
                <h3 className="mb-2 text-xl">あなたの回答</h3>
                {currentAnswer[index].content}
              </div>
            </>
          )}
          <ModelAnswerContent
            project_id={project_id}
            question_id={question.id}
          />
        </div>
      ))}
    </>
  );
};
