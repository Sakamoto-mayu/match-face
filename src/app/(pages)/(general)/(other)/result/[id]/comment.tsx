import TextArea from "@/components/ui/TextArea";
import { Comment } from "@/const/result";
import { User } from "@/const/result";
import Image from "next/image";

type Props = {
  user_id: number;
  project_id: number;
};

export const CommentContent = ({ user_id, project_id }: Props) => {
  const currentComment = Comment.filter(
    (comment) => comment.user_id === user_id && project_id === project_id
  );
  const commentUser = User.filter(
    (user) => user.id === currentComment[0].admin_id
  );

  return (
    <>
      <h3 className="text-xl mb-3">{commentUser[0].name}からのコメント</h3>
      <div className="border border-black rounded-md p-3 flex justify-between ">
        <div className="w-1/6 flex justify-center items-center">
          <Image
            src={commentUser[0].image_url}
            alt="コメントしたユーザーのアイコン"
            width={70}
            height={70}
            priority
          />
        </div>
        <div id="comment" className="w-5/6">
          {currentComment[0].content}
        </div>
      </div>
    </>
  );
};
