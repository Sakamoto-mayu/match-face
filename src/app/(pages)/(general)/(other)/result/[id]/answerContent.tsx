"use client";

import { project } from "@/const/testing";
import { AnswerList } from "./answerList";
import { CommentContent } from "./comment";
import OrangeButton from "@/components/ui/button/OrangeButton";
import { Answer } from "@/const/result";
import { useRouter } from "next/navigation";

type Props = {
  user_id: number;
  project_id: number;
};

export const AnswerContent = ({ user_id, project_id }: Props) => {
  const router = useRouter();
  const currentProject = project.filter(
    (project) => project.id === Number(project_id)
  );
  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === Number(project_id)
  );

  return (
    <div className="w-1/2 py-5">
      <div className="flex justify-between items-center my-3">
        <h1 className="text-2xl">回答の確認</h1>
        {/*現状のデータでは4以降の解答データを呼び出そうとするとエラーになります */}
        <p>回答日時:{currentAnswer[0].answer_date.toLocaleDateString()}</p>
      </div>
      <hr className="border border-black" />
      <h2 className="text-xl my-2">案件名：{currentProject[0].name}</h2>
      <p className="text-lg">{currentProject[0].project_detail}</p>
      <AnswerList project_id={project_id} />
      <CommentContent user_id={user_id} project_id={project_id} />
      <div className="flex justify-around my-20">
        <OrangeButton
          className="border rounded-none"
          label="質問一覧へ"
          onClick={() => router.push("/questions")}
        />
        <OrangeButton
          className="border rounded-none"
          label="回答履歴一覧へ"
          onClick={() => router.push("/histories")}
        />
      </div>
    </div>
  );
};
