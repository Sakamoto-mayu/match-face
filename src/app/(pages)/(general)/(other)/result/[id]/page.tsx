import { AnswerContent } from "./answerContent";

const ResultPage = ({ params }: { params: { id: number } }) => {
  const user_id = 1;

  return (
    <div className="flex flex-col items-center">
      <AnswerContent user_id={user_id} project_id={params.id} />
    </div>
  );
};

export default ResultPage;
