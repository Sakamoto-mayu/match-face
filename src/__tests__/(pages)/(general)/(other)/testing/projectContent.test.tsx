import { ProjectContent } from "@/app/(pages)/(general)/(other)/testing/[id]/projectContent";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { project } from "@/const/testing";
import "@testing-library/jest-dom";

const user = userEvent.setup();

const testData = {
  id: 1,
};
const currentProject = project.filter((project) => project.id === testData.id);

describe("ProjectContentコンポーネントのテスト", () => {
  test("案件名と案件詳細が表示される", () => {
    render(<ProjectContent id={testData.id} />);
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toHaveTextContent(`案件名:${currentProject[0].name}`);

    const project_detail = document.querySelector("#project_detail");
    expect(project_detail).toHaveTextContent(
      `${currentProject[0].project_detail}`
    );
  });
  test("下書きボタンをクリックするとモーダルが開き、キャンセルボタンをクリックするとモーダルが閉じる", async () => {
    render(<ProjectContent id={testData.id} />);
    //ボタンの存在を確認
    const button = screen.getByRole("button", { name: "下書きを保存する" });
    expect(button).toBeInTheDocument();
    // ボタンをクリックする前はモーダルが開いていないことを確認
    const modal_close = document.querySelector("#draft-modal");
    expect(modal_close).not.toBeInTheDocument();
    //ボタンをクリック
    await user.click(button);
    //モーダルが開いたことを確認
    const modal_open = document.querySelector("#draft-modal");
    expect(modal_open).toBeInTheDocument();
    //モーダル内のキャンセルボタンをクリック
    const modal_button = screen.getByRole("button", { name: "キャンセル" });
    await user.click(modal_button);
    // モーダルが閉じたことを確認
    const modal_clicked = document.querySelector("#draft-modal");
    expect(modal_clicked).not.toBeInTheDocument();
  });
  test("回答送信ボタンをクリックするとモーダルが開き、キャンセルボタンをクリックするとモーダルが閉じる", async () => {
    render(<ProjectContent id={testData.id} />);
    //ボタンの存在を確認
    const button = screen.getByRole("button", { name: "回答を送信する" });
    expect(button).toBeInTheDocument();
    // ボタンをクリックする前はモーダルが開いていないことを確認
    const modal_close = document.querySelector("#answer-modal");
    expect(modal_close).not.toBeInTheDocument();
    //ボタンをクリック
    await user.click(button);
    // モーダルが開いたことを確認
    const modal_open = document.querySelector("#answer-modal");
    expect(modal_open).toBeInTheDocument();
    //モーダル内のキャンセルボタンをクリック
    const modal_button = screen.getByRole("button", { name: "キャンセル" });
    await user.click(modal_button);
    //モーダルが閉じたことを確認
    const modal_clicked = document.querySelector("#answer-modal");
    expect(modal_clicked).not.toBeInTheDocument();
  });
  test("スナップショットテスト", () => {
    const { container } = render(<ProjectContent id={testData.id} />);
    expect(container).toMatchSnapshot();
  });
});
