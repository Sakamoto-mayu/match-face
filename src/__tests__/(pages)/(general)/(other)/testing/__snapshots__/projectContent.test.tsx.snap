// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContentコンポーネントのテスト スナップショットテスト 1`] = `
<div>
  <div
    class="w-1/2 py-5"
  >
    <h1
      class="text-xl my-2"
    >
      案件名:
      案件のタイトルが入ります
    </h1>
    <p
      class="text-lg"
      id="project_detail"
    >
      案件の詳細内容が入ります。テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
    </p>
    <div
      class="py-5"
    >
      <h2>
        Q
        1
        ：
        質問のタイトルが入ります
      </h2>
      <p
        class="mt-2 mb-4"
      >
        テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
      </p>
      <label
        for="質問のタイトルが入ります"
      >
        <p>
          回答記入欄
        </p>
      </label>
      <textarea
        class="border border-black mt-1 w-full"
        cols="50"
        id="質問のタイトルが入ります"
        name="質問のタイトルが入ります"
        rows="6"
      />
    </div>
    <div
      class="py-5"
    >
      <h2>
        Q
        2
        ：
        質問のタイトルが入ります
      </h2>
      <p
        class="mt-2 mb-4"
      >
        テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
      </p>
      <div>
        <input
          id="choice0"
          name="質問のタイトルが入ります"
          type="radio"
          value="選択肢A"
        />
        <label
          for="choice0"
        >
          選択肢A
        </label>
      </div>
      <div>
        <input
          id="choice1"
          name="質問のタイトルが入ります"
          type="radio"
          value="選択肢B"
        />
        <label
          for="choice1"
        >
          選択肢B
        </label>
      </div>
      <div>
        <input
          id="choice2"
          name="質問のタイトルが入ります"
          type="radio"
          value="選択肢C"
        />
        <label
          for="choice2"
        >
          選択肢C
        </label>
      </div>
      <div>
        <input
          id="choice3"
          name="質問のタイトルが入ります"
          type="radio"
          value="選択肢D"
        />
        <label
          for="choice3"
        >
          選択肢D
        </label>
      </div>
    </div>
    <div
      class="py-5"
    >
      <h2>
        Q
        3
        ：
        質問のタイトルが入ります
      </h2>
      <p
        class="mt-2 mb-4"
      >
        テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
      </p>
      <label
        for="質問のタイトルが入ります"
      >
        <p>
          回答記入欄
        </p>
      </label>
      <textarea
        class="border border-black mt-1 w-full"
        cols="50"
        id="質問のタイトルが入ります"
        name="質問のタイトルが入ります"
        rows="6"
      />
    </div>
    <div
      class="flex justify-around my-20"
    >
      <div>
        <button
          class="bg-neutral-300 drop-shadow-lg hover:brightness-95 active:drop-shadow-none active:shadow-inner active:mt-0.5 text-black p-2 h-auto text-lg border rounded-none w-50"
        >
          下書きを保存する
        </button>
      </div>
      <div>
        <button
          class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white p-2 w-40 h-auto text-lg border rounded-none"
        >
          回答を送信する
        </button>
      </div>
    </div>
  </div>
</div>
`;
